File: Ejercicio3/app/src/main/java/com/example/ejercicio3/MainActivity.java
package com.example.ejercicio3;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    RadioGroup rgOpciones;
    RadioButton rbDolares, rbSoles;
    Button btnConvertir;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        // Enlazamos con el XML
        rgOpciones = findViewById(R.id.rgOpciones);
        rbDolares = findViewById(R.id.rbDolares);
        rbSoles = findViewById(R.id.rbSoles);
        btnConvertir = findViewById(R.id.btnConvertir);

        btnConvertir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Aquí iría la lógica de conversión
                // Por ahora, mostraremos qué opción se eligió
                if (rbDolares.isChecked()) {
                    Toast.makeText(MainActivity.this, "Seleccionó Dólares", Toast.LENGTH_SHORT).show();
                } else {
                    Toast.makeText(MainActivity.this, "Seleccionó Soles", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: Ejercicio1/app/src/main/java/com/example/ejercicio1_prac2/MainActivity.kt
package com.example.ejercicio1_prac2
import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
class MainActivity : AppCompatActivity() {
        private val tasasCambio = mapOf(
        "Dólar (USD)" to 1.0,
        "Sol Peruano (PEN)" to 3.42,
        "Euro (EUR)" to 0.86,
        "Libra (GBP)" to 0.75,
        "Rupia (INR)" to 92.87,
        "Real (BRL)" to 5.22,
        "Peso (MXN)" to 17.75,
        "Yuan (CNY)" to 6.90,
        "Yen (JPY)" to 158.31
    )
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val etMonto = findViewById<EditText>(R.id.etMonto)
        val spinnerOrigen = findViewById<Spinner>(R.id.spinnerOrigen)
        val spinnerDestino = findViewById<Spinner>(R.id.spinnerDestino)
        val btnConvertir = findViewById<Button>(R.id.btnConvertir)

        // Configurar los Spinners con la lista de strings.xml
        val adaptador = ArrayAdapter.createFromResource(
            this,
            R.array.monedas_nombres,
            android.R.layout.simple_spinner_item
        )
        adaptador.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinnerOrigen.adapter = adaptador
        spinnerDestino.adapter = adaptador
        btnConvertir.setOnClickListener {
            val montoStr = etMonto.text.toString()

            if (montoStr.isNotEmpty()) {
                val monto = montoStr.toDouble()
                val monedaDe = spinnerOrigen.selectedItem.toString()
                val monedaA = spinnerDestino.selectedItem.toString()

                // 1. Convertir moneda de origen a USD base
                val montoEnDolares = monto / tasasCambio[monedaDe]!!

                // 2. Convertir de USD a moneda de destino
                val resultadoFinal = montoEnDolares * tasasCambio[monedaA]!!

                val mensaje = "$monto de $monedaDe equivalen a ${"%.2f".format(resultadoFinal)} $monedaA"
                Toast.makeText(this, mensaje, Toast.LENGTH_LONG).show()
            } else {
                Toast.makeText(this, "Por favor ingresa un monto", Toast.LENGTH_SHORT).show()
            }
        }
    }
}